packaging/debian/control

```diff
@@ -18,7 +18,7 @@ Description: Ansible IT Automation
 
 Package: ansible-test
 Architecture: all
-Depends: ansible (= ${binary:Version}), python-virtualenv, ${misc:Depends}
+Depends: ansible-base (= ${binary:Version}), python3-virtualenv, ${misc:Depends}
```

Comment:
We probably want `python3-venv` instead of `python3-virtualenv`, although in most cases `ansible-test` can use either one.

Reply:
It looks like python3-venv just includes the /usr/bin/pyvenv executable.
Do we need that or does ansible-test work without hte exectuable file?

Also i seem to recall that Debian or Ubuntu had modified ensurepip which
then broke venv. So we might need virtualenv to work around that:

https://bugs.launchpad.net/ubuntu/+source/python3.4/+bug/1290847

Reply:
No, we don't need the executable, just the ability to use `python -m venv`.

At least on Ubuntu we need `python3-venv` to be able to use `python -m venv`, but I don't know if that's a Ubuntu customization or something inherited from Debian.

Reply:
Hmm... It is the same on both Debian and Ubuntu:
* https://packages.ubuntu.com/groovy/amd64/python3-venv/filelist
* https://packages.debian.org/buster/amd64/python3-venv/filelist
( You can click around to see if the packaging has changed in different releases but from what I saw they're all the same filelist)

I wonder if this is related to venv being broken on Ubuntu (and maybe Debian?) in some way?

Reply:
Last time I checked, using `-m venv` on Ubuntu failed with an error message unless `python3-venv` was installed, since the functionality had been split out into a separate package.

Reply:
Still seems broken:
```
root@dfd16c0396df:/# cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

root@dfd16c0396df:/# apt install python3-virtualenv
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  python3-distutils python3-lib2to3 python3-pkg-resources
Suggested packages:
Preparing to unpack .../python3-distutils_3.6.9-1~18.04_all.deb ...
Unpacking python3-distutils (3.6.9-1~18.04) ...
Selecting previously unselected package python3-pkg-resources.
Preparing to unpack .../python3-pkg-resources_39.0.1-2_all.deb ...
Unpacking python3-pkg-resources (39.0.1-2) ...
Selecting previously unselected package python3-virtualenv.
Preparing to unpack .../python3-virtualenv_15.1.0+ds-1.1_all.deb ...
Unpacking python3-virtualenv (15.1.0+ds-1.1) ...
Setting up python3-pkg-resources (39.0.1-2) ...
Setting up python3-lib2to3 (3.6.9-1~18.04) ...
Setting up python3-distutils (3.6.9-1~18.04) ...
Setting up python3-virtualenv (15.1.0+ds-1.1) ...
root@dfd16c0396df:/# python3 -m virtualenv testenv
Running virtualenv with interpreter /usr/bin/python2
New python executable in /testenv/bin/python2
Also creating executable in /testenv/bin/python
Installing setuptools, pkg_resources, pip, wheel...done.

root@dfd16c0396df:/# python3 -m venv testenv2
The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt-get install python3-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.

Failing command: ['/testenv2/bin/python3', '-Im', 'ensurepip', '--upgrade', '--default-pip']
```

Really odd since I don't see what in the `python3-venv` package matters to `ensurepip`. Something weird going on there.

Reply:
I got the same behavior on Debian 10, so it seems this is something that's inherited from Debian.

Reply:
I think the error message makes it pretty clear that on Debian/Ubuntu you must install `python3-venv` to use `ensurepip` or `-m venv`.

Reply:
Oh, it's not python3-venv..... It's python3.x-venv (which contains the ensurepip library rather than the venv library.).  python3-venv has a dependency on the python3.x-venv package.  Depending on the python3-venv package instead the 3.x package directly probably lets the package work when Debian/Ubuntu update their default python3 as well so even though we don't need the pyvenv script, I guess depending on the unversioned package is the best thing to do?

I am a little worried about this comment (from the end of the bug I posted earlier... It seem that like launchpad doesn't let you link to the comment in the stream of the bug):

https://bugs.launchpad.net/ubuntu/+source/python3.4/+bug/1290847/comments/107

But I've played around with the pyhton3-venv and python3.6-venv package just like samdoran has now and I can't find a way to replicate that (or see how it would be replicable given the package contents).

Reply:
+1 from me to change the dep to python3-venv in case that wasn't obvious from my last comment.